pulsebot/bugzilla.py

```diff
@@ -11,91 +11,59 @@
 
 
 class BugzillaError(Exception):
+    def __init__(self, message):
+        logger.exception(message)
     pass
 
 
 class Bugzilla(object):
     def __init__(self, server, api_key):
         self._server = server.rstrip("/")
-        self._session = requests.Session()
-        self._session.headers.update({"User-Agent": "pulsebot"})
-        self._session.headers.update({"X-Bugzilla-API-Key": api_key})
-
-    def get_fields(self, bug, fields):
-        bug_url = "%s/rest/bug/%d?include_fields=%s" % (
-            self._server,
-            bug,
-            ",".join(fields),
-        )
-        logger.info(f"get_fields: {bug_url}")
-        try:
-            r = self._session.get(bug_url)
-            r.raise_for_status()
-            bug_data = r.json()
-        except Exception:
-            logger.exception(
-                f"Error occurred retrieving bug fields {bug_url} {r.status_code}"
-            )
-            raise BugzillaError()
-
+        self._headers = {
+            "User-Agent": "pulsebot",
+            "X-Bugzilla-API-Key": api_key,
+            "Content-Type": "application/json",
+            "Accept": "application/json"
+        }
+
+    def _call(self, method, path, data=None):
+        data = data or {}
+        method = method or 'GET'
+        bug_url = f"{self._server}/{path}"
+        logger.info(f"{method}: {bug_url}")
+        if method.lower() == 'get':
+            r = requests.get(bug_url, headers=self._headers)
+        elif method.lower() == 'post':
+            r = requests.post(bug_url, data=data, headers=self._headers)
+        elif method.lower() == 'PUT':
+            r = requests.put(bug_url, data=data, headers=self._headers)
```

Comment:
nit: You could try:

```python
if method.lower() == 'get':
    requests.get(...)
elif method.lower() in ('put', 'post'):
    getattr(requests, method)(bug_url, data=data, headers=self._headers)
else:
    raise BugzillaError(f"Unknown method: {method} {bug_url}")
...
```

Since `POST` and `PUT` share the same parameters.